### ISSUE TITLE
Full page reload when hmr fails (trigger automatically)

### ISSUE THREAD

**Issue by kachkaev** [mxstbr/react-boilerplate#1353]
Just a quick addition to the repo's knowledge base since someone may search for the same problem in the issues.

This boilerplate uses `webpack-hot-middleware`, which attempts to reload the contents of the react components on fly when they are changed. Quite often, however, it gets stuck and you see this message:

```
[HMR] The following modules couldn't be hot updated: (Full reload needed)
This is usually because the modules which have changed (and their parents) do not know how to hot reload themselves. See http://webpack.github.io/docs/hot-module-replacement-with-webpack.html for more details.
```

This requires to focus on the browser and manually start refresh, which can become frustrating during a series of micro-changes.

When an app persists its data, a full refresh does not break its state so can be triggered automatically without much harm. To do so go to `internals/webpack/webpack.dev.babel.js`, find `'webpack-hot-middleware/client',` and replace with `'webpack-hot-middleware/client?reload=true'`. This and other options are explained here: https://github.com/glenjamin/webpack-hot-middleware#config

After you restart your app's server, any change will trigger a refresh, no matter if it is hot or full. Yay!

**Comment by mxstbr** [comment]
We should do this by default. Would you mind submitting a PR with this change?

**Comment by gihrig** [comment]
Implemented in #1357 - Closing.
